tmt/steps/execute/__init__.py

```diff
@@ -904,11 +836,17 @@ def extract_results(
                 invocation=invocation,
                 default_log=invocation.relative_path / TEST_OUTPUT_FILENAME)
 
-        # Handle the 'tmt-report-result' command results as a single test
+        # Load the results from the `tmt-report-results.yaml` if a file was generated.
+        tmt_report_results = []
         if self._tmt_report_results_filepath(invocation).exists():
-            return self.extract_tmt_report_results(invocation)
+            tmt_report_results = self.extract_tmt_report_results(invocation)
 
-        return invocation.test.test_framework.extract_results(invocation, logger)
+        # Propagate loaded `tmt_report_results` to test framework, which will handle these results
+        # accordingly (e.g. saves them as a tmt subresults).
+        return invocation.test.test_framework.extract_results(
+            invocation=invocation,
+            results=tmt_report_results,
+            logger=logger)
```

Comment:
Are the parameter names necessary? The call might fit on one line without them (and with a small change of `tmt_report_results` becoming `reported_results`, or even just `results`? :)

Reply:
Sure, modified. Thanks :)